### PR TITLE
zapi: add squash command

### LINKED ISSUE #2651: zapi: Use zed lake commands
The zapi command should use the same code as zed lake. We will have to add additional api routes to extend functionality of zed lake serve in order to make this happen. The zed lake command will also have to use some interface for lake.Root and lake.Pool so this can be switched out with the api client for zapi.

### PR DESCRIPTION
**Pull request by mattnibs**
Add squash endpoint to service

Closes #2651

### CODE REVIEW

File: cmd/zed/lake/squash/command.go [hunk 1]
```diff
@@ -32,6 +33,7 @@ respect to the head of the pool's commit journal at the time it is run.
 
 func init() {
 	zedlake.Cmd.Add(Squash)
+	zedapi.Cmd.Add(Squash)
```
Comment (nwt) [inline cmd/zed/lake/squash/command.go:36]:
```suggestion
	zedapi.Cmd.Add(Squash)
	zedlake.Cmd.Add(Squash)
```

File: service/core.go [hunk 2]
```diff
@@ -145,6 +145,7 @@ func (c *Core) addAPIServerRoutes() {
 	c.authhandle("/pool/{pool}/staging", handleScanStaging).Methods("GET")
 	c.authhandle("/pool/{pool}/staging/{commit}", handleCommit).Methods("POST")
 	c.authhandle("/pool/{pool}/stats", handlePoolStats).Methods("GET")
+	c.authhandle("/pool/{pool}/squash", handleSquash).Methods("POST")
```
Comment (nwt) [inline service/core.go:148]:
```suggestion
	c.authhandle("/pool/{pool}/squash", handleSquash).Methods("POST")
	c.authhandle("/pool/{pool}/staging", handleScanStaging).Methods("GET")
	c.authhandle("/pool/{pool}/staging/{commit}", handleCommit).Methods("POST")
	c.authhandle("/pool/{pool}/stats", handlePoolStats).Methods("GET")
```

File: service/handlers.go [hunk 3]
```diff
@@ -340,6 +340,27 @@ func handleDelete(c *Core, w *ResponseWriter, r *Request) {
...
+	}
+	pool, err := c.root.OpenPool(r.Context(), id)
+	if err != nil {
+		w.Error(err)
+		return
+	}
+	var req api.SquashRequest
+	if !r.Unmarshal(w, &req) {
+		return
+	}
```
Comment (nwt) [inline service/handlers.go:356]:
```suggestion
	var req api.SquashRequest
	if !r.Unmarshal(w, &req) {
		return
	}
	pool, err := c.root.OpenPool(r.Context(), id)
	if err != nil {
		w.Error(err)
		return
	}
```

File: service/ztests/squash.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+script: |
+  source service.sh
+  zapi create -q -p POOL
+  a=$(zapi add -p POOL a.zson | head -1 | awk '{print $1}')
+  b=$(zapi add -p POOL b.zson | head -1 | awk '{print $1}')
+  id=$(zapi squash -p POOL $a $b | head -1 | awk '{print $NF}')
+  zapi commit -q -p POOL $id
+  zapi query -z "from POOL | sort ."
```
Comment (nwt) [inline service/ztests/squash.yaml:8]:
```suggestion
  zapi create -q -p test
  a=$(zapi add -p test a.zson | head -1 | awk '{print $1}')
  b=$(zapi add -p test b.zson | head -1 | awk '{print $1}')
  id=$(zapi squash -p test $a $b | head -1 | awk '{print $NF}')
  zapi commit -q -p test $id
  zapi query -z "from test | sort ."
```

**Review by nwt — APPROVED** [review]

File: service/handlers.go (hunk 3 above)
Comment (mattnibs) [reply service/handlers.go:356]: Fair.

**Pull request merged**